src/permissions/permissionsManager.ts

```diff
@@ -0,0 +1,124 @@
+import roles from './roles';
+import _get from 'lodash/get';
+import { jwtDecode } from 'jwt-decode';
+
+interface Role {
+  title: string;
+  permissions: string[];
+}
+
+const fallbackRole = 'GUEST';
+
+/**
+ * This function takes a the JWT token of the user and a permission, and validates if the user is allowed to perform the action associated with the permission.
+ * @param {string} token A JWT token string from Firestore User.
+ * @param {string} permission The permission to check.
+ * @returns {Boolean} True if the user has the permission, false otherwise.
+ */
+export function canUser(token: string, permission: string): boolean {
+  const userRole: string = getRoleFromToken(token);
+  const role: Role = _get(roles, userRole.toUpperCase());
+
+  // Check to see if the user's role has the requested permission
+  const roleTitle: string = userRole.toUpperCase();
+  if (roleTitle === 'SUPER_ADMIN') return true;
+  const roleHasPermission = checkPermissionList(role.permissions, permission);
+  if (roleHasPermission) return true;
+
+  // Else, return false
+  return false;
+}
+
+/**
+ * This function takes a JWT token and returns the user's role. If the token is invalid or missing the role claim,
+ * return the GUEST role.
+ *
+ * @param {string} token JWT token string from Firestore User.
+ * @returns {string} The user's role based on the provided JWT token.
+ */
+function getRoleFromToken(token: string): string {
+  const decodedToken = jwtDecode(token);
+
+  // Retrieve the user's role from the token's claims. If the claim is missing or invalid, default to the GUEST role.
+  const userRole: string = _get(decodedToken, 'role', fallbackRole);
+  if (!_get(decodedToken, 'role')) {
+    console.warn(
+      `[ROAR Permissions Manager] Invalid or missing role claim in User's custom claims. Defaulting to the ${fallbackRole} role.`,
+    );
+  }
+  return userRole;
```

Comment:
This could be slightly simplified as well, and same as above, without needing `lodash`. The decoded JWT token should, ideally, also have a proper typing

```suggestion
  const decodedToken = jwtDecode<DecodedToken>(token)
  const userRole = decodedToken.role ?? fallbackRole
  
  if (!decodedToken.role) {
    console.warn(
      `[ROAR Permissions Service] Missing role claim in User's custom claims. Defaulting to the ${fallbackRole} role.`
    )
  }
  
  return userRole
```